docs/snapshots.rst

```diff
@@ -91,3 +93,33 @@ they next poll our mutable directory.
 We cannot sign local Snapshots because we don't yet have the
 capability-string for the content. The capability-string is only known
 after we upload the content to Tahoe-LAFS.
+
+
+Signature Details
+-----------------
+
+As above, we upload the content and metadata as two distinct immutable
+capability. There is also the "mangled" name of the file. This mangled
+name will also point at the Snapshot from the Personal DMD.
+
+The signature scheme is to concatenate all three of the above
+(separated by newlines) and then produce a signature over it using the
+author's signing key (which is a `nacl.signing.SigningKey` from the
+PyNaCl library (which uses `libsodium` under the hood). See
+https://pynacl.readthedocs.io/en/latest/signing/
+
+The resulting signature is base32-encoded and included in the "tahoe
+metadata" for the "metadata capability" entry in the Snapshot's
+immutable directory.
+
+To verify the signature, a client has the content and metadata
+capability-strings when they download the Snapshot object and the
+mangled name from that participant's Personal DMD. They can thus
+verify the signature using the participant's public-key before
+downloading the content or metadata.
+
+**Note**: "metadata" is used in two ways above; Tahoe allows you to
+add arbitrary metadata for each entry in a directory. This is where
+we locate the base32-encoded signature itself. The "metadata about
+the Snapshot" is its own capability (consisting of JSON-encoded
+metadata).
```

Comment:
I didn't realize until very near the end of the diff but it seems like we should specify what "metadata about the Snapshot" consists of, exactly.  Since we're relying on the signature to validate the metadata for us, we should be sure everything we need to validate about a  snapshot (except for content) is included in the "metadata about the Snapshot".

Noted elsewhere it may make sense to include the parent capabilities in this metdata.  Also, maybe the path should be part of the metadata as well, instead of a third top-level-ish thing.